### PR TITLE
Update gitops & configsync readmes

### PR DESCRIPTION
**Pull request by karlkfi**

### CODE REVIEW

File: catalog/gitops/configsync/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Resources
+
+```
+File                    APIVersion                         Kind               Name                                        Namespace
+config-management.yaml  configmanagement.gke.io/v1         ConfigManagement   config-management                           
+configsync-iam.yaml     iam.cnrm.cloud.google.com/v1beta1  IAMPolicyMember    source-reader-sync-cluster-name-project-id  config-control
+configsync-iam.yaml     iam.cnrm.cloud.google.com/v1beta1  IAMPolicyMember    sync-cluster-name                           config-control
+configsync-iam.yaml     iam.cnrm.cloud.google.com/v1beta1  IAMServiceAccount  sync-cluster-name                           config-control
+setters.yaml            v1                                 ConfigMap          setters
```
Comment (bharathkkb) [inline catalog/gitops/configsync/README.md:27]: This won't/shouldn't be applied

File: catalog/gitops/README.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
-# CSR Git Ops Pipeline
+# GitOps package
```
Comment (bharathkkb) [inline catalog/gitops/README.md:1]: Suggest we keep CSR here since this blueprint only works with CSRs?

Comment (karlkfi) [reply catalog/gitops/README.md:1]: It's using the name from the Kptfile, which ideally should match the name of the parent directory, according to kpt docs.

@morgante renamed the parent package to remove the csr detail and there was a discussion about CSR being the default impl (other impls would need an identifier in the name). So I named this child package accordingly.

If we want CSR in the name, it should be in the directory name and the kptfile.

File: catalog/gitops/configsync/README.md (hunk 1 above)
Comment (karlkfi) [reply catalog/gitops/configsync/README.md:27]: The script to generate this doesn't ignore local-config=true files. Should it?

Also, most of the other setter.yaml files in this branch do not have the local-config=true annotation. So they are actually applied...

Comment (karlkfi) [reply catalog/gitops/configsync/README.md:27]: I'm putting the script in a different PR.

Comment (bharathkkb) [reply catalog/gitops/configsync/README.md:27]: [kptv1 will filter out fnconfig](https://github.com/GoogleContainerTools/kpt/blob/2a817f60cf7132c88fd2e526c02b800cf927c048/pkg/live/load_test.go#L70-L92) defined in the kptfile if applied via `kpt live apply`

File: catalog/gitops/README.md (hunk 2 above)
Comment (morgante) [reply catalog/gitops/README.md:1]: I don't agree that the name of the directory needs to match the name in the README. The README name should be more descriptive and full.

Please revert.

File: catalog/gitops/README.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
...
+
+GitOps CI/CD pipeline for use with ACM.
+
+After installing this blueprint, you will be able to:
 
-This blueprint instantiates a GitOps CI/CD pipeline using Cloud Source Repositories and Cloud Build. After installing this blueprint, you will be able to:
```
Comment (morgante) [inline catalog/gitops/README.md:3]: Why was this removed? It's important that CSR usage is stated up front.

File: catalog/gitops/configsync/README.md (hunk 1 above)
Comment (morgante) [reply catalog/gitops/configsync/README.md:27]: Yeah, kpt filters out the function configs. Function configs should not be listed with actual resources.

File: catalog/gitops/README.md (hunk 2 above)
Comment (karlkfi) [reply catalog/gitops/README.md:1]: > The directory rename was based on your suggestion. 

I did suggest shortening the old `csr-git-ops-pipeline` name, and you shortened it to `csr-gitops`, which is what's on main. 

After you had already merged that, I suggested using `gitops-csr` instead, to set a precedence to make others for -github and -gitlab and they would still sort next to each other alphabetically. But you shortened it to `gitops` instead for kpt-v1.

Perhaps you misunderstood my suggestion, but my recollection is that you explicitly did not want to use the -csr suffix because you wanted to make the GCP-native path the default.

I'll put the fact that it uses CSR back in the description, but I don't think it belongs in the title of the readme unless it's in the name of the package. This readme was manually modified, but I have a script to generate them directly from the kptfile, in which case it would be `# gitops package`.

Comment (morgante) [reply catalog/gitops/README.md:1]: Yes, I chose to remove the `-csr` suffix so this can be the default.

The name for packages in the readme does not need (and probably should not) match the package name. In the directory name, we often prioritize brevity. the README title should prioritize descriptiveness.

Comment (morgante) [reply catalog/gitops/README.md:1]: Just to re-emphasize the point, *most* of our packages use titles in the README which are different (more verbose) than the directory name. For example: https://github.com/GoogleCloudPlatform/blueprints/tree/main/catalog/bucket

There is absolutely no requirement for the titles to match and they should not. `google-cloud-storage-bucket` would be an excessively long directory name while `bucket page` would be a poor title for the package.

File: catalog/gitops/README.md (hunk 3 above)
Comment (karlkfi) [reply catalog/gitops/README.md:3]: Removed on accident while trying to be one concise. I'll add it back. I didn't mean to hide the detail that it uses CSR from the description.

File: catalog/gitops/README.md (hunk 2 above)
Comment (karlkfi) [reply catalog/gitops/README.md:1]: > A human-readable description is not the same as a directory name.

I agree, and I think the name in the Kptfile should be human readable. If it were, then pulling it out to use for the readme title would be fine. But kpt docs and code say that they should be the same.

If we want them to be different, to facilitate readme generation, then we should ask the kpt team to change it. We can of course manually modify titles in the mean time, but then we can't use the readme generator for validation in pre-commit tests.

Comment (morgante) [reply catalog/gitops/README.md:1]: go/blueprint-metadata calls for a different "name" (machine-oriented) vs "title" (human-oriented) which has not yet been implemented. I agree we should continue pushing for inclusion of `title` metadata.

For the time being, let's just manually modify titles. We will have to make the README generator be able to tolerate some discrepancies anyways.

**Comment by karlkfi** [comment]
Obsolete. Replaced by https://github.com/GoogleCloudPlatform/blueprints/pull/49
